### ISSUE TITLE
Better understand time of periastron passage

### ISSUE THREAD

**Issue by Rodrigo-Tenorio** [PyFstat/PyFstat#137]
After doing some more informative plots using #135 , it looks like there's something odd related to the inference of time of periastron passage.

![PyFstat_example_semi_coherent_binary_search_using_MCMC_walkers](https://user-images.githubusercontent.com/49152330/91310455-2dfd6e80-e7b2-11ea-8e56-79cd8785d31d.png)

**Comment by Rodrigo-Tenorio** [comment]
Things don't look to improve if we add some leeway into the periastron argument...
![PyFstat_example_semi_coherent_binary_search_using_MCMC_walkers](https://user-images.githubusercontent.com/49152330/91312010-04454700-e7b4-11ea-9714-c91348348a78.png)

**Comment by Rodrigo-Tenorio** [comment]
High eccentricity doesn't look to change the game:
![PyFstat_example_semi_coherent_binary_search_using_MCMC_walkers](https://user-images.githubusercontent.com/49152330/91312985-24c1d100-e7b5-11ea-93f7-c679d133e678.png)

**Comment by dbkeitel** [comment]
I know almost nothing about how tp is implemented, so guessing completely in the dark: is there any chance this may have something to do with reference time offset between injection and recovery?

**Comment by Rodrigo-Tenorio** [comment]
> I know almost nothing about how tp is implemented, so guessing completely in the dark: is there any chance this may have something to do with reference time offset between injection and recovery?

My guesses right now are either that or a different convetion: tp is essentially an initial phase (it's actually `tp * Omega`), so you can either choose [0, 2pi) or [-pi, pi). BinarySkyHough chooses the latter, and I'd say [LALSuite ](https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/group___s_s_btimes__h.html#gac88f72b4d1e7281a9bb1f0bbf131bf3a) somehow chooses the former...

tl;dr It may well just be a matter of conventions

**Comment by Rodrigo-Tenorio** [comment]
Just checked a different `tref`: No changes.

In fact, if I'm not mistaken, `tref` is consistently used as the reference point to Taylor-expand the phase evolution of a continuous wave in terms of the `source emission time` i.e. it's defined before taking into account any proper motion (the only affected parameters are the initial phase, the frequency and spin-down parameters).

**Comment by dbkeitel** [comment]
That should be it, yes! I just thought about it because I know we set a `tref=tstart` default somewhere, while most lal codes actually default to `tref=tmid`.

**Comment by Rodrigo-Tenorio** [comment]
I've extended the `tp` prior to a full orbital period above and below the injected value, but that doesn't seem to improve the result. I'll dig into Fstat codes to see how it's actually implemented then.
![PyFstat_example_semi_coherent_binary_search_using_MCMC_walkers](https://user-images.githubusercontent.com/49152330/91333245-a02f7c80-e7cd-11ea-97d6-f18ee2e08713.png)

**Comment by dbkeitel** [comment]
The offset looks too big to be just any sort of MCMC issue in itself, but you could still also try a grid search in tp only, say from 1.002 to 1.005 (a bit narrower than the initial prior), keeping all other parameters fixed, to see if it puts the peak at the same offset, and how it's shaped.

**Comment by Rodrigo-Tenorio** [comment]
Indeed, it doesn't look like an MCMC issue to me. 

![PyFstat_example_semi_coherent_binary_search_using_MCMC_walkers](https://user-images.githubusercontent.com/49152330/91359294-910ff500-e7f4-11ea-96cc-c62d88b44da4.png)

![PyFstat_example_semi_coherent_binary_search_using_MCMC_corner](https://user-images.githubusercontent.com/49152330/91359530-04b20200-e7f5-11ea-9075-c1b7fda01f4d.png)

**Comment by Rodrigo-Tenorio** [comment]
I just realized I forgot to do the obvious check:

Zero eccentricity and zero periastron argument:
![PyFstat_example_semi_coherent_binary_search_using_MCMC_walkers](https://user-images.githubusercontent.com/49152330/91453330-dfb6a100-e87f-11ea-9dd3-0be1afa17d88.png)
![PyFstat_example_semi_coherent_binary_search_using_MCMC_corner](https://user-images.githubusercontent.com/49152330/91453344-e34a2800-e87f-11ea-8d81-feb6ea649b3c.png)

**Comment by Rodrigo-Tenorio** [comment]
Zero eccentricity, pi / 4 periastron argument. 

![Uploading PyFstat_example_semi_coherent_binary_search_using_MCMC_walkers.png…]()

**Comment by Rodrigo-Tenorio** [comment]
Staying in zero eccentricity an periastron argument, setting tighter priors:

![PyFstat_example_semi_coherent_binary_search_using_MCMC_walkers](https://user-images.githubusercontent.com/49152330/91566305-be1eed80-e943-11ea-803a-ee629c5461a2.png)

I'd say this definitely closes the discussion on whether this is an MCMC issue or a definition of `tp`, as the obtained Fstat values are significantly smaller than in previous cases.

**Comment by Rodrigo-Tenorio** [comment]
@dbkeitel , seems that #120 has more ramifications than expected.

Here I present a first plot, done using the binary example in 63c1aecf822e1f74c37a42e18d888e74a5c85731
![PyFstat_example_semi_coherent_binary_search_using_MCMC_prior_posterior](https://user-images.githubusercontent.com/49152330/91580908-10fda280-e94e-11ea-8282-c02320a4ad2a.png)

On the other hand, here is a plot using the very same example, this time in 86fec75f4b451d1d2078b4b9b358c1699fdd93d9

![PyFstat_example_semi_coherent_binary_search_using_MCMC_walkers](https://user-images.githubusercontent.com/49152330/91581046-3d192380-e94e-11ea-8a16-04f64ad89b17.png)

I'll check the rest of parsing functions in MFD to make sure nothing else is broken.

**Comment by dbkeitel** [comment]
Thanks for the thorough investigation! Fully fixed with #143?

**Comment by Rodrigo-Tenorio** [comment]
Indeed, fixed with #143
